src/main/java/io/elastic/sailor/Service.java

```diff
@@ -46,32 +46,67 @@ public Service(ComponentResolver resolver,
 
 
     public static void main(String[] args) throws IOException {
-        if (args.length < 1) {
-            throw new IllegalArgumentException("1 argument is required, but were passed " + args.length);
-        }
 
-        Injector injector = Guice.createInjector(new ServiceModule(), new ServiceEnvironmentModule());
+        try {
+
+            if (args.length < 1) {
+                throw new IllegalArgumentException("1 argument is required, but were passed " + args.length);
+            }
+
+            Injector injector = Guice.createInjector(new ServiceModule(), new ServiceEnvironmentModule());
+
+            final Service service = injector.getInstance(Service.class);
 
-        final Service service = injector.getInstance(Service.class);
+            final String methodName = args[0];
 
-        final String methodName = args[0];
+            logger.info("Starting execution of {}", methodName);
 
-        logger.info("Starting execution of {}", methodName);
+            final ServiceMethods method = ServiceMethods.parse(methodName);
 
-        final ServiceMethods method = ServiceMethods.parse(methodName);
+            service.executeMethod(method);
 
-        service.start(method);
+        } catch (Exception e) {
 
+            processException(e);
+        }
     }
 
-    public void start(final ServiceMethods method) throws IOException {
+    public void executeMethod(final ServiceMethods method) {
+        final JsonObject data = method.execute(this.params);
+
+        JsonObject payload = new JsonObject();
+        payload.addProperty("status", "success");
+        payload.add("data", data);
+
+        sendData(this.postResultUrl, payload);
+    }
 
-        final JsonObject result = method.execute(this.params);
+    private static void processException(Exception e) {
+
+        JsonObject data = new JsonObject();
+        data.addProperty("message", e.getMessage());
+
+        JsonObject payload = new JsonObject();
+        payload.addProperty("status", "success");
+        payload.add("data", data);
+
+        sendData(Utils.getOptionalEnvVar(Constants.ENV_VAR_POST_RESULT_URL), payload);
+    }
+
+    private static void sendData(String url, JsonObject payload) {
 
         logger.info("Sending response");
 
-        String response = Utils.postJson(this.postResultUrl, result);
+        if (url == null) {
```

Comment:
this url can't be null. The validation is done by guice. we can remove this check